internal/plugins/helm/init.go

```diff
@@ -0,0 +1,212 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package helm
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/prometheus/common/log"
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/pkg/scaffold"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	cmdutil "github.com/operator-framework/operator-sdk/internal/plugins/helm/internal"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/helm"
+	"github.com/operator-framework/operator-sdk/internal/scaffold/input"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+)
+
+type initPlugin struct {
+	// For help text.
+	commandName string
+
+	// Helm flags
+	APIVersion       string
+	Kind             string
+	CRDVersion       string
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+	gitInit          bool
+	SkipGeneration   bool
+}
+
+var (
+	_ plugin.Init        = &initPlugin{}
+	_ cmdutil.RunOptions = &initPlugin{}
+)
+
+func (p *initPlugin) UpdateContext(ctx *plugin.Context) {
+	ctx.Description = `Initialize a Helm project with the Helm Chart package directories.`
+	ctx.Examples = fmt.Sprintf(`  # Scaffold a Helm project %s 
+	
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --api-version=app.example.com/v1alpha1 \
+  --kind=AppService \
+  --helm-chart=myrepo/app
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=myrepo/app
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=myrepo/app \
+  --helm-chart-version=1.2.3
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=app \
+  --helm-chart-repo=https://charts.mycompany.com/
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=app \
+  --helm-chart-repo=https://charts.mycompany.com/ \
+  --helm-chart-version=1.2.3
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=/path/to/local/chart-directories/app/
+
+  $ operator-sdk init --plugins=helm.kubebuilder.io/v1-alpha \
+  --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
+	"
+`,
+		ctx.CommandName)
+
+	p.commandName = ctx.CommandName
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&p.APIVersion, "api-version", "",
```

Comment:
If we decide to scaffold an API on `init --plugins helm` then this should be split into `--group` and `--version`.

Reply:
done the suggested change. I agree.

Reply:
Also, see that currently, we will break the API version in the group and etc as well after the user inform. See here for an example: https://github.com/operator-framework/operator-sdk/blob/424a61d56000e6e3d91d352faa1bd4f7c814661f/internal/scaffold/resource.go#L120

Reply:
You can do
```go
scaffold.NewResource(path.Join(p.Group+"."+p.Domain, p.Version), p.Kind)
```

Realistically we should create a resource from [options](https://github.com/estroz/operator-sdk/commit/ec96d37d71478bf4d596560739b6dd806f2ce355#diff-6a0b4bad7f4daebf727110c716686c43R71-R103), like kubebuilder does it, for flexibility and backwards compatibility.

Reply:
It is what is in my mind for the next steps.